pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+			fmt.Sprintf("Failed to write defaults to egress gateway Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	// Set the condition to progressing
+	setProgressing(r.client, ctx, egw, string(operatorv1.ResourceNotReady), fmt.Sprintf("Name = %s, Namespace = %s", egw.Name, egw.Namespace))
+
+	egwVxlanPort := egressgateway.DefaultEGWVxlanPort
+	egwVxlanVNI := egressgateway.DefaultEGWVxlanVNI
+	if fc.Spec.EgressIPVXLANPort != nil {
+		egwVxlanPort = *fc.Spec.EgressIPVXLANPort
+	}
+	if fc.Spec.EgressIPVXLANVNI != nil {
+		egwVxlanVNI = *fc.Spec.EgressIPVXLANVNI
+	}
+
+	openshift := r.provider == operatorv1.ProviderOpenShift
+	config := &egressgateway.Config{
+		PullSecrets:       pullSecrets,
+		Installation:      installation,
+		OSType:            rmeta.OSTypeLinux,
+		EgressGW:          egw,
+		EgressGWVxlanPort: egwVxlanPort,
+		EgressGWVxlanVNI:  egwVxlanVNI,
+		Openshift:         openshift,
+		UsePSP:            r.usePSP,
+	}
+
+	component := egressgateway.EgressGateway(config)
+	ch := utils.NewComponentHandler(log, r.client, r.scheme, egw)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		reqLogger.Error(err, "Error with images from ImageSet")
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		setDegraded(r.client, ctx, egw, reconcileErr,
+			fmt.Sprintf("Error with images from ImageSet Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	if err = ch.CreateOrUpdateOrDelete(ctx, component, r.status); err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error creating / updating resource: Name = %s, Namespace = %s", egw.Name, egw.Namespace))
+		r.status.SetDegraded(operatorv1.ResourceUpdateError,
+			fmt.Sprintf("Error creating / updating resource: Name = %s, Namespace = %s", egw.Name, egw.Namespace), err, reqLogger)
+		setDegraded(r.client, ctx, egw, reconcileErr,
+			fmt.Sprintf("Error creating / updating resource Name = %s, Namespace = %s, err = %s", egw.Name, egw.Namespace, err.Error()))
+		return reconcile.Result{}, err
+	}
+
+	// Update the status of this CR.
+	egw.Status.State = operatorv1.TigeraStatusReady
+	setAvailable(r.client, ctx, egw, string(operatorv1.AllObjectsAvailable), "All objects available")
+
+	// After the resource is created/updated, TigeraStatus needs to be set by taking the cumulative status of the
+	// available EGW resources. Lets say we create 2 EGW resources Red, Blue. Both are degraded. Now lets create 3rd resource
+	// yellow. Though yellow is reconciled and rendered successfully, TigeraStatus should still be degraded as Red, Blue are
+	// degraded. Now lets assume Blue gets updated and gets rendered properly. In this case, TigeraStatus should still be
+	// degraded as Red is unhealthy. "status" represents, the cumulative status of EGWs.
+	if unreadyEGW != nil {
```

Comment:
I don't think this should be done inside of this function - we call this for every EGW that exists in the cluster, and there's no point in checking the same single unreadyEGW for all of them.

We should handle unreadyEGWs separately.

I'm not sure why we need to handle unready EGWs in the controller at all - the status manager is meant to handle that itself by monitoring deployment statuses - is this logic even needed?